### PR TITLE
Add declarative Gradle upgrade recipes

### PR DESCRIPTION
**Pull request by timtebeek**
## What's your motivation?
When upgrading Java versions folks also need to upgrade older Gradle versions. While upgrading the wrapper is one part of that upgrade, at times there's required changes as well, in particular going from 4.x to 5.0. Some of these changes we can already automate, others might still follow now that the framework for these migrations is there.

## Anything in particular you'd like reviewers to focus on?
- For the `RemoveEnableFeaturePreview` record I went by when those keys were dropped from the enum, going by latest.patch; seemed ok, but could be double checked as documentation was hard to find.
- Intentionally left out `compile` -> `api/implementation`, although that'll be a bigger pain point. Anything we can do for reasonable coverage here? Perhaps look if `java-library` plugin is defined and base our behavior on that? Or at worst merely post a comment on the first case?

## Have you considered any alternatives or workarounds?
The components here were already available. We could document the steps and considerations for `compile` -> `api/implementation`, and how they could be achieved with recipes.

## Any additional context
- https://github.com/openrewrite/rewrite/issues/4194

### CODE REVIEW

**Comment by shanman190** [comment]
General question, but are these more migration recipes and as such should follow the [migration ADR](https://github.com/openrewrite/rewrite/blob/main/doc%2Fadr%2F0004-library-migration-conventions.md)?

**Comment by timtebeek** [comment]
Good call! Wasn't aware of that doc, but helps to standardize where possible.

Any other steps that come to mind? Things like required common plugin upgrades, or removal of jcenter?

Of course we can expand the steps later as well; just let me know when you'd consider this ready for a first merge.

**Comment by shanman190** [comment]
So the `jcenter()` method is still available as of Gradle 8.10.1, but with that said JFrog has redirected https://jcenter.bintray.com/ to https://repo1.maven.org/maven2/. So in practice it's now the same as `mavenCentral()`. So I could go either way there.

File: rewrite-gradle/src/main/resources/META-INF/rewrite/gradle-6.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  - org.openrewrite.gradle.ChangeDependencyConfiguration:
+      groupId: "*"
+      artifactId: "*"
+      configuration: testCompile
+      newConfiguration: testImplementation
+  - org.openrewrite.gradle.ChangeDependencyConfiguration:
+      groupId: "*"
+      artifactId: "*"
+      configuration: runtime
+      newConfiguration: runtimeOnly
```
Comment (shanman190) [inline rewrite-gradle/src/main/resources/META-INF/rewrite/gradle-6.yml:37]: supernit: In practice this doesn't change the meaning, but generally when I'm thinking about these sort of changes, I'm thinking in terms of `main` and then `test`.

```suggestion
  - org.openrewrite.gradle.ChangeDependencyConfiguration:
      groupId: "*"
      artifactId: "*"
      configuration: runtime
      newConfiguration: runtimeOnly
  - org.openrewrite.gradle.ChangeDependencyConfiguration:
      groupId: "*"
      artifactId: "*"
      configuration: testCompile
      newConfiguration: testImplementation
```

**Review by shanman190 — APPROVED** [review]

**Pull request merged**